src/config/mongoose_config_parser_toml.erl

```diff
@@ -518,30 +519,707 @@ process_module([Mod|_] = Path, Opts) ->
+    [{connections, Conns}];
+module_opt([<<"cache">>, <<"mod_global_distrib">>|_] = Path, V) ->
+    Cache = parse_section(Path, V),
+    [{cache, Cache}];
+module_opt([<<"bounce">>, <<"mod_global_distrib">>|_], false) ->
+    [{bounce, false}];
+module_opt([<<"bounce">>, <<"mod_global_distrib">>|_] = Path, V) ->
+    Bounce = parse_section(Path, V),
+    [{bounce, Bounce}];
+module_opt([<<"redis">>, <<"mod_global_distrib">>|_] = Path, V) ->
+    Redis = parse_section(Path, V),
+    [{redis, Redis}];
+module_opt([<<"hosts_refresh_interval">>, <<"mod_global_distrib">>|_], V) ->
+    [{hosts_refresh_interval, V}];
+module_opt([<<"proxy_host">>, <<"mod_jingle_sip">>|_], V) ->
+    [{proxy_host, b2l(V)}];
+module_opt([<<"proxy_port">>, <<"mod_jingle_sip">>|_], V) ->
+    [{proxy_port, V}];
+module_opt([<<"listen_port">>, <<"mod_jingle_sip">>|_], V) ->
+    [{listen_port, V}];
+module_opt([<<"local_host">>, <<"mod_jingle_sip">>|_], V) ->
+    [{local_host, b2l(V)}];
+module_opt([<<"sdp_origin">>, <<"mod_jingle_sip">>|_], V) ->
+    [{sdp_origin, b2l(V)}];
+module_opt([<<"ram_key_size">>, <<"mod_keystore">>|_], V) ->
+    [{ram_key_size, V}];
+module_opt([<<"keys">>, <<"mod_keystore">>|_] = Path, V) ->
+    Keys = parse_list(Path, V),
+    [{keys, Keys}];
+module_opt([<<"archive_chat_markers">>, <<"mod_mam_meta">>|_], V) ->
+    [{archive_chat_markers, V}];
+module_opt([<<"pm">>, <<"mod_mam_meta">>|_], false) ->
+    [];
+module_opt([<<"pm">>, <<"mod_mam_meta">>|_] = Path, V) ->
+    PM = parse_section(Path, V),
+    [{pm, PM}];
+module_opt([<<"muc">>, <<"mod_mam_meta">>|_], false) ->
+    [];
+module_opt([<<"muc">>, <<"mod_mam_meta">>|_] = Path, V) ->
+    Muc = parse_section(Path, V),
+    [{muc, Muc}];
+module_opt([<<"extra_lookup_params">>, <<"mod_mam_meta">>|_], V) ->
+    [{extra_lookup_params, b2a(V)}];
+module_opt([_, <<"mod_mam_meta">>|_] = Path, V) ->
+    mod_mam_opts(Path, V);
+module_opt([<<"host">>, <<"mod_muc">>|_], V) ->
+    [{host, b2l(V)}];
+module_opt([<<"access">>, <<"mod_muc">>|_], V) ->
+    [{access, b2a(V)}];
+module_opt([<<"access_create">>, <<"mod_muc">>|_], V) ->
+    [{access_create, b2a(V)}];
+module_opt([<<"access_admin">>, <<"mod_muc">>|_], V) ->
+    [{access_admin, b2a(V)}];
+module_opt([<<"access_persistent">>, <<"mod_muc">>|_], V) ->
+    [{access_persistent, b2a(V)}];
+module_opt([<<"history_size">>, <<"mod_muc">>|_], V) ->
+    [{history_size, V}];
+module_opt([<<"room_shaper">>, <<"mod_muc">>|_], V) ->
+    [{room_shaper, b2a(V)}];
+module_opt([<<"max_room_id">>, <<"mod_muc">>|_], <<"infinity">>) ->
+    [{max_room_id, infinity}];
+module_opt([<<"max_room_id">>, <<"mod_muc">>|_], V) ->
+    [{max_room_id, V}];
+module_opt([<<"max_room_name">>, <<"mod_muc">>|_], <<"infinity">>) ->
+    [{max_room_name, infinity}];
+module_opt([<<"max_room_name">>, <<"mod_muc">>|_], V) ->
+    [{max_room_name, V}];
+module_opt([<<"max_room_desc">>, <<"mod_muc">>|_], <<"infinity">>) ->
+    [{max_room_desc, infinity}];
```

Comment:
It's `infinite` not `infinity`, as I stated in the comments before :-) Please change the parsed atom.

Reply:
Right, it is a bit misleading having these 2 types of "infinity" though

Reply:
That's why I think it's best to change only the resulting atom and maybe get rid of this second infinity later.